src/Xamarin.Android.Build.Tasks/Utilities/MarshalMethodsAssemblyRewriter.cs

```diff
@@ -23,32 +23,24 @@ sealed class AssemblyImports
 
 		IDictionary<string, IList<MarshalMethodEntry>> methods;
 		ICollection<AssemblyDefinition> uniqueAssemblies;
-		IDictionary <string, HashSet<string>> assemblyPaths;
+		IDictionary<AssemblyDefinition, string> assemblyPaths;
 		TaskLoggingHelper log;
 
-		public MarshalMethodsAssemblyRewriter (IDictionary<string, IList<MarshalMethodEntry>> methods, ICollection<AssemblyDefinition> uniqueAssemblies, IDictionary <string, HashSet<string>> assemblyPaths, TaskLoggingHelper log)
+		public MarshalMethodsAssemblyRewriter (IDictionary<string, IList<MarshalMethodEntry>> methods, ICollection<AssemblyDefinition> uniqueAssemblies, IDictionary<AssemblyDefinition, string> assemblyPaths, TaskLoggingHelper log)
 		{
+			this.assemblyPaths = assemblyPaths;
 			this.methods = methods ?? throw new ArgumentNullException (nameof (methods));
 			this.uniqueAssemblies = uniqueAssemblies ?? throw new ArgumentNullException (nameof (uniqueAssemblies));
-			this.assemblyPaths = assemblyPaths ?? throw new ArgumentNullException (nameof (assemblyPaths));
```

Comment:
So we're saying that `assemblyPaths` *can* be `null` now?  Or are we relying on nullable reference types?

I'm confused; as far as I can tell, `$(Nullable)` is *not* set in `Xamarin.Android.Build.Tasks`, yet other lines *in this file* use nullable reference types.  How is that working?

(*Other* files in this project use `#nullable enable`, but not *this* file, so *how* is it using nullable reference types?)

Reply:
Don't pay attention to `MarshalMethodsAssemblyRewriter` in this PR, it's going to change anyway later.